plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -536,10 +568,22 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 handle.getSchema(),
                 handle.getIndex(),
                 newDomain,
+                newRegexes,
                 handle.getQuery(),
                 handle.getLimit());
 
-        return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), false));
+        return Optional.of(new ConstraintApplicationResult<>(handle, TupleDomain.withColumnDomains(unsupported), newExpression, false));
+    }
+
+    private String likeToRegexp(String like)
+    {
+        String regexp = like.replaceAll("\\\\", "\\\\\\\\"); // first, escape regexp's escape character
+        for (char c : REGEXP_RESERVED_CHARACTERS) {
+            regexp = regexp.replaceAll("\\" + c, "\\\\" + c);
+        }
+        return regexp
+                .replaceAll("%", ".*")
+                .replaceAll("_", ".");
```

Comment:
how often LIKE pattern is used with ESCAPE character?
we should follow up with adding support for that. ideally we should keep the ESCAPE characters variety out of connectors and do some kind of normalization on the engine side.

let's have some TODO comment in places like here

Reply:
AFAIU from [the documentation](https://trino.io/docs/current/functions/comparison.html#pattern-comparison-like), ESCAPE character is used in order to escape `_` and `%`. In addition, any character can be the ESCAPE character.
Added a TODO in the engine + a TODO here that we should support ESCAPE character when it's supported by the engine.

Reply:
FWIW, Trino JDBC driver generates queries (over `system.jdbc`) always using the `ESCAPE '\'`.
so eg tools may be doing that.

> when it's supported by the engine.

engine does support escape. wdym here?

Reply:
> FWIW, Trino JDBC driver generates queries (over system.jdbc) always using the ESCAPE '\\'.
so eg tools may be doing that.

Do you want me to block connectors from using '\\' as an escape character? how?

> engine does support escape. wdym here?

The engine doesn't pushdown escape (yet), so we can't support it here before that

Reply:
> Do you want me to block connectors from using '\' as an escape character? how?

No. Just saying that i expect ESCAPE to be in use in real life.
TODO is fine for now.